### PR TITLE
Project Samples: Select all

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Adds the ability to select/deselect all samples in a project.

1. Added a checkbox to the first header that, when checked, selects all samples in the project 
2. Added the selection count and total samples to the table footer.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

- Go to a project that has samples (e.g. [Outbreak 2023](http://localhost:3000/yersinia/yersinia-pseudotuberculosis/outbreak-2023/-/samples))
- Click the select all checkbox, all samples should be selected
- Click the select all checkbox again, all samples should be de-selected
- Check to make sure the table footer is updated with the correct number of samples.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/javascript/controllers/selection_controller.js [hunk 1]
```diff
@@ -81,4 +96,13 @@ export default class extends Controller {
       outlet.setDisabled(count);
     });
   }
+
+  #updateSelectAllCB(total) {
```
Comment (deepsidhu85) [inline app/javascript/controllers/selection_controller.js:100]: Can I get you to update this method name for clarity?

File: test/system/projects/samples_test.rb [hunk 2]
```diff
@@ -1911,6 +1911,68 @@ class SamplesTest < ApplicationSystemTestCase
...
+      end
+
+      find('input[data-action="filters#submit"]').send_keys([:control, 'a'], :space)
+      within 'tfoot' do
+        assert_selector 'strong[data-selection-target="total"]', text: '3'
+        assert_selector 'strong[data-selection-target="selected"]', text: '0'
+      end
+    end
+
     def retrieve_puids
```
Comment (deepsidhu85) [inline test/system/projects/samples_test.rb:1976]: Can I get you to also add a test for de-selecting when all are selected?

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks great and working as expected. Just have a few comments below

File: app/controllers/projects/samples_controller.rb [hunk 3]
```diff
@@ -113,6 +114,21 @@ def destroy # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
       end
     end
 
+    def select
+      authorize! @project, to: :sample_listing?
+
+      respond_to do |format|
+        format.turbo_stream do
+          if params[:select].present?
+            @q = load_samples.ransack(params[:q])
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/controllers/projects/samples_controller.rb:123]: Updated in 05cf9baf6b0ae6c187e8f8332432f4da401744bb

File: app/javascript/controllers/selection_controller.js (hunk 1 above)
Comment (joshsadam) [reply app/javascript/controllers/selection_controller.js:100]: Updated in bd016c88ca3108a6bbf14b9c63fb2bb9bba44bdd

File: test/system/projects/samples_test.rb (hunk 2 above)
Comment (joshsadam) [reply test/system/projects/samples_test.rb:1987]: Added in 3060cdd

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**